core/pkg/resultshandling/reporter/v2/mockreporter_test.go

```diff
@@ -83,3 +139,219 @@ func TestReportMock_strToDisplay(t *testing.T) {
+		o.ResourceAttackTracks[k] = val
+	}
+
+	o.AttackTracks = make(map[string]v1alpha1.IAttackTrack, len(v.AttackTracks))
+	for k, val := range v.AttackTracks {
+		o.AttackTracks[k] = val
+	}
+
+	return &o
+}
+
+func (s *testServer) Root() string {
+	return s.Server.URL
+}
+
+func (s *testServer) URL(pth string) string {
+	pth = strings.TrimLeft(pth, "/")
+
+	return fmt.Sprintf("%s/%s", s.Server.URL, pth)
+}
+
+// mockAPIServer builds a mock API running with a TLS endpoint.
+//
+// Running tests with the DEBUG_TEST=1 environment will result in dumping a trace of
+// the incoming requests.
+func mockAPIServer(t testing.TB) *testServer {
+	h := http.NewServeMux()
+
+	server := &testServer{
+		Server: httptest.NewUnstartedServer(h),
+	}
+
+	h.HandleFunc(pathTestReport, func(w http.ResponseWriter, r *http.Request) {
+		if os.Getenv("DEBUG_TEST") != "" {
+			dump, _ := httputil.DumpRequest(r, true)
+			t.Logf("%s\n", dump)
+		}
+
+		if !assert.Equal(t, http.MethodPost, r.Method) {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+
+			return
+		}
+
+		if !assert.NoErrorf(t, r.ParseForm(), "expected params to parse") {
+			w.WriteHeader(http.StatusBadRequest)
+
+			return
+		}
+
+		cluster := r.Form.Get("clusterName")
+		contextName := r.Form.Get("contextName")
+		customer := r.Form.Get("customerGUID")
+		report := r.Form.Get("reportGUID")
+
+		if cluster == "" || contextName == "" || customer == "" || report == "" {
+			t.Error("missing query parameter")
+			w.WriteHeader(http.StatusBadRequest)
+
+			return
+		}
+
+		// NOTE(fredbi): (i)  requests should have header Content-Type: "application/json"
+		// NOTE(fredbi): (ii) shouldn't we require an extra authentication (e.g. secretKey or Token)?
+
+		buf, err := io.ReadAll(r.Body)
+		defer func() {
+			_ = r.Body.Close()
+		}()
+
+		if !assert.NoError(t, err) {
+			w.WriteHeader(http.StatusInternalServerError)
+
+			return
+		}
+
+		var input reporthandlingv2.PostureReport
+		if !assert.NoError(t, json.Unmarshal(buf, &input)) {
+			w.WriteHeader(http.StatusInternalServerError)
+
+			return
+		}
+	})
+
+	h.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		dump, _ := httputil.DumpRequest(r, true)
+		t.Logf("%s\n", dump)
+
+		t.Errorf("unexpected route in input request: %v", r.URL)
+
+		w.WriteHeader(http.StatusNotFound)
+	})
+
+	server.StartTLS()
+
+	return server
+}
+
+// newInterceptor builds a new http.RoundTripper to re-route outgoing requests.
+func newInterceptor(transport http.RoundTripper, host string) *interceptor {
+	return &interceptor{
+		original: transport,
+		host:     host,
+	}
+}
+
+func (i *interceptor) RoundTrip(r *http.Request) (*http.Response, error) {
+	defer r.Body.Close()
+
+	hijacked := r.Clone(r.Context())
+	hijacked.URL.Host = i.host
+
+	return i.original.RoundTrip(hijacked)
+}
+
+// hijackedClient builds an HTTP client suited for working against a mock server.
+//
+// This client supports mocked TLS and re-routes outgoing calls to the local mock server.
```

Comment:
nice trick, I didn't know this would work with TLS

Reply:
yes I discovered that too :)